### ISSUE TITLE
configure check for __res_ndestroy() generated wrong value on CentOS 8.3

### ISSUE THREAD

**Issue by bicknell** [trusteddomainproject/OpenDMARC#141]
On my CentOS 8.3 box the configure script sets:

```
/* Define to 1 if you have the `__res_ndestroy()' function. */
#define HAVE_RES_NDESTROY 1
```

However CentOS 8.3 does not have that function.  I looked into the configure check and I'm unsure why it's getting the wrong value, so I don't have a patch.  The code compiles file by manually changing that one #define to #undef.  

I have a hunch though.  There are two search calls in configure.ac that set the _same variable_:

```
AC_SEARCH_LIBS(res_ndestroy, resolv,
        AC_DEFINE(HAVE_RES_NDESTROY, 1,
        [Define to 1 if you have the `res_ndestroy()' function.]))

AC_SEARCH_LIBS(__res_ndestroy, resolv,
        AC_DEFINE(HAVE_RES_NDESTROY, 1,
        [Define to 1 if you have the `__res_ndestroy()' function.]))
```

Note that the second one actually searches for the one proceeded by double underscores.  I wonder if these two functions operating on the same variable is somehow confusing autoconf.

**Comment by bicknell** [comment]
Ah ha!  I was able to track down the problem, but I'm not yet sure the proper cross platform solution.

I kindly direct you to https://github.com/trusteddomainproject/OpenDMARC/blob/master/configure.ac#L147.  Note that someone has used `AC_LANG_PROGRAM` to define the prolog and body of the test function.  This body checks for `res_ninit`.  However, the program is not reset until https://github.com/trusteddomainproject/OpenDMARC/blob/master/configure.ac#L162.  That means the _same program body_ is being used for the checks for `res_ninit`, `res_ndestroy`, `__res_ninit`, and `__res_ndestroy`.  Basically if `res_ninit` exists it will always set them all to true, otherwise always set them all to false.  I was able to confirm this much by moving the reset in line 162 to right after the `res_ninit` check.

I am unsure if that's good enough on it's own because of the comment above.  I'm thinking one or more of these may require including `resolv.conf` like the `res_ninit` check, and if that's the case it would require 4 different `AC_LANG_PROGRAM` calls, one for each check.  Maybe only some of them need checks though.  In any event it will require some testing across multiple platforms to be sure the logic is correct.

I suspect this will currently cause a build failure on any system that has res_ninit`, but is missing one of `res_ndestroy`, `__res_ninit`, or `__res_ndestroy`.

**Comment by thegushi** [comment]
@martinbogo is our tester for most linux platforms.  Can you have a test build and let me know what you spot?  I'm so loathe to change any more of autoconf, but I definitely want this to work.

**Comment by thegushi** [comment]
My thought is that this was fixed in the autoconf changes we made in testing and develop, as spotted by @gene-git.  I've pinged leo privately to see if the problem still occurs to him on those branches.

**Comment by gene-git** [comment]
I can confirm that this issue is fixed in develop/testing.

**Comment by thegushi** [comment]
Awesome.  Imma close it out.  Leo can reopen if he finds otherwise.

**Comment by bicknell** [comment]
Problem is fixed for me in the develop branch on CentOS 8.3.
